support for ordinal forecasts

elray1 opened the pull request:
This PR adds support for ordinal forecasts, mainly consisting of the ranked probability score.

There is still the potential for numerical issues [as outlined in this discussion](https://github.com/hubverse-org/hubEvals/discussions/64).  However, after I switched to using data frames of scores from hubExamples rather than copies of those stored as csv, the unit tests are passing.  So I propose to merge these updates in and address the numerical issues separately.

annakrystalli reviewed (approved):
This generally looks good to me. My only minor concern is how we are communicating where a user might find the order of ordinal categories required by `output_type_id_order`. I know this issue exists: https://github.com/hubverse-org/hubUtils/issues/153 and will go a long way towards accessing this info and ideally being able to be used to extract it automatically by default. Until then though, I think it would help to provide some guidance in the form of documentation for this argument.

elray1 commented:
Thanks for the review, @annakrystalli !  I addressed your individual comments on files directly, and in b632535 I added a sentence to function docs noting that the output type id orders could be looked up in the hub's config file. I'm actually not too worried about this, though, because in cases where the order matters, I think it will typically be clear to users what the order should be -- e.g., "low", "moderate", "high", "very high".
